pkg/network/render.go

```diff
@@ -185,15 +213,17 @@ func RenderAdditionalNetworks(conf *netv1.NetworkConfigSpec, manifestDir string)
 		return nil, errors.Errorf("invalid Additional Network Configuration: %v", errs)
 	}
 
-	// render Multus when additional networks is provided
-	if len(ans) > 0 {
-		objs, err := renderMultusConfig(manifestDir)
+	if len(ans) == 0 {
+		return nil, nil
+	}
+
+	// render the CRD when additional networks are provided
+	if !netAttachDefCRDRendered {
```

Comment:
This can't happen. We only allow additional networks if multus is enabled, and if multus is enabled then either it already rendered the NetworkAttachmentDefinition CRD or else the renderer errored out before reaching this point.